ext/opentelemetry-ext-flask/src/opentelemetry/ext/flask/__init__.py

```diff
@@ -29,9 +29,10 @@
 
 .. code-block:: python
 
-    from opentelemetry.ext.flask import FlaskInstrumentor
-    FlaskInstrumentor().instrument()  # This needs to be executed before importing Flask
     from flask import Flask
+    from opentelemetry.ext.flask import FlaskInstrumentor
+
+    Flask = FlaskInstrumentor().instrument(flask_class=Flask)
```

Comment:
This looks too complicated to me. The `instrument` method does not do any instrumentation but just returns an instrumented version of the Flask class that the **user has** to assign to the local Flask reference. Btw, the current `uninstrument` will do nothing after that call, somehow the user will have to update the `Flask` reference to the original `flask.Flask` class.

Reply:
This leads me to believe we might need to patch the flask object instead of the class itself (like a middleware)? This way we will have direct control over what is instrumented or not. What if the user wants only some flask instances to be instrumented and others not to be? And same goes for uninstrumenting as well.